geos: update GEOS to 3.12.3

yuzefovich opened the pull request:
Most of the test changes are due to ordering difference within POLYGONs (which seems ok). Others were compared against PostGIS running GEOS 3.13.1 and mostly match Postgres. The following differences were found:
- a couple cases in TestSimplifyGEOS appear to be close enough
- `ST_LineInterPolatePoints` now returns an error for `LINESTRING EMPTY`, which wasn't the case before for us with 3.11.2, nor is the case for PostGIS with 3.13.1.

I spent some time trying to install PostGIS 3.4.3 locally which uses GEOS 3.12.x, but brew makes it hard, so I gave up. I plan to bump GEOS to 3.13.1 shortly, so hopefully some of the differences will be reverted.

Epic: None
Release note (general change): Spatial libraries from Cockroach now rely on GEOS 3.12 instead of GEOS 3.11.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/148859)
<!-- Reviewable:end -->

yuzefovich committed:
geos: update GEOS to 3.12.3

Most of the test changes are due to ordering difference within POLYGONs
(which seems ok). Others were compared against PostGIS running GEOS
3.13.1 and mostly match Postgres. The following differences were found:
- a couple cases in TestSimplifyGEOS appear to be close enough
- `ST_LineInterPolatePoints` now returns an error for `LINESTRING
EMPTY`, which wasn't the case before for us with 3.11.2, nor is the
case for PostGIS with 3.13.1.

I spent some time trying to install PostGIS 3.4.3 locally which uses
GEOS 3.12.x, but brew makes it hard, so I gave up. I plan to bump GEOS
to 3.13.1 shortly, so hopefully some of the differences will be
reverted.

Release note (general change): Spatial libraries from Cockroach
now rely on GEOS 3.12 instead of GEOS 3.11.

rickystewart reviewed (approved):
The remaining test failure seems unrelated?

yuzefovich commented:
friendly ping @cockroachdb/sql-queries-prs

mgartner reviewed (approved):
LGTM
